localstack/services/s3/s3_listener.py

```diff
@@ -681,7 +681,9 @@ def return_response(self, method, path, data, headers, response):
 
                 # fix content-type: https://github.com/localstack/localstack/issues/618
                 #                   https://github.com/localstack/localstack/issues/549
-                if 'text/html' in response.headers.get('Content-Type', ''):
+                #                   https://github.com/localstack/localstack/issues/854
+                if 'text/html' in response.headers.get('Content-Type', '') \
+                        and not response_content_str.startswith('<!doctype html'):
```

Comment:
should be a case insensitive match though no? DOCTYPE and doctype are both widely used

Reply:
@whummer as per https://html.spec.whatwg.org/multipage/syntax.html#the-doctype